File: src/main/java/io/playground/orderservice/infrastructure/client/PaymentFeignClient.java
package io.playground.orderservice.infrastructure.client;

import io.playground.orderservice.application.saga.dto.ClientDto;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient(
        name = "paymentClient",
        url = "${api.internal.payment-service.url}"
)
public interface PaymentFeignClient {
    @PostMapping("/approve")
    ResponseEntity<Void> approvePayment(@RequestBody ClientDto.ApprovePaymentRequest request);

    @PostMapping("/cancel")
    ResponseEntity<Void> cancelPayment(@RequestBody ClientDto.CancelPaymentRequest request);

    @PostMapping("/partial-cancel")
    ResponseEntity<Void> cancelPartially(@RequestBody ClientDto.CancelPartiallyRequest request);
}

File: src/main/kotlin/io/playground/orderservice/infrastructure/client/PaymentFeignClient.kt
package io.playground.orderservice.infrastructure.client

import io.playground.orderservice.application.saga.dto.ClientDto
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody

@FeignClient(
    name = "paymentClient",
    url = "\${api.internal.payment-service.url}",
)
interface PaymentFeignClient {
    @PostMapping("/approve")
    fun approvePayment(@RequestBody request: ClientDto.ApprovePaymentRequest): ResponseEntity<Void>

    @PostMapping("/cancel")
    fun cancelPayment(@RequestBody request: ClientDto.CancelPaymentRequest): ResponseEntity<Void>

    @PostMapping("/partial-cancel")
    fun cancelPartially(@RequestBody request: ClientDto.CancelPartiallyRequest): ResponseEntity<Void>
}
